core/src/test/java/org/elasticsearch/index/get/GetFieldTests.java

```diff
@@ -0,0 +1,192 @@
+        mutations.add(() -> new GetField(randomUnicodeOfCodepointLength(15), getField.getValues()));
+        mutations.add(() -> new GetField(getField.getName(), randomGetField(XContentType.JSON).v1().getValues()));
+        return randomFrom(mutations).get();
+    }
+
+    public static Tuple<GetField, GetField> randomGetField(XContentType xContentType) {
+        if (randomBoolean()) {
+            String fieldName = randomFrom(ParentFieldMapper.NAME, RoutingFieldMapper.NAME, UidFieldMapper.NAME);
+            GetField getField = new GetField(fieldName, Collections.singletonList(randomAsciiOfLengthBetween(3, 10)));
+            return Tuple.tuple(getField, getField);
+        }
+        String fieldName = randomAsciiOfLengthBetween(3, 10);
+        Tuple<List<Object>, List<Object>> tuple = randomStoredFieldValues(xContentType);
+        GetField input = new GetField(fieldName, tuple.v1());
+        GetField expected = new GetField(fieldName, tuple.v2());
+        return Tuple.tuple(input, expected);
+    }
+
+    //TODO move this to some utility class
+    /**
+     * Asserts that the provided {@link BytesReference}s hold the same content. The comparison is done between the map
+     * representation of the provided objects.
+     */
+    public static void assertSameOutput(BytesReference expected, BytesReference actual, XContentType xContentType) throws IOException {
+        //we tried comparing byte per byte, but that didn't fly for a couple of reasons:
+        //1) whenever anything goes through a map, ordering is not preserved, which is perfectly ok
+        //2) Jackson SMILE parser parses floats as double, which then get printed out as double (with double precision)
+        try (XContentParser parser = xContentType.xContent().createParser(actual)) {
+            Map<String, Object> finalMap = parser.map();
+            try (XContentParser parser2 = xContentType.xContent().createParser(expected)) {
+                Map<String, Object> originalMap = parser2.map();
+                assertEquals(originalMap, finalMap);
+            }
+        }
+    }
+
+    //TODO move this to some utility class? It should be useful in other tests too, whenever we parse stored fields
+    /**
+     * Returns a tuple containing random stored field values and their corresponding expected values once printed out
+     * via {@link ToXContent#toXContent(XContentBuilder, ToXContent.Params)} and parsed back via {@link XContentParser#objectText()}.
+     * Generates values based on what can get printed out. Stored fields values are retrieved from lucene and converted via
+     * {@link org.elasticsearch.index.mapper.MappedFieldType#valueForDisplay(Object)} to either strings, numbers or booleans.
+     *
+     * @param xContentType the content type, used to determine what the expected values are for float numbers.
+     */
+    public static Tuple<List<Object>, List<Object>> randomStoredFieldValues(XContentType xContentType) {
```

Comment:
I think it can be private for now and the TODO removed... If really needed outside of this test, we could maybe create a `RandomObjects` utilities class in the test framework and add all this kind of method there.